test/unit/org/apache/cassandra/index/sai/cql/BaseDataModel.java

```diff
@@ -176,7 +176,7 @@ public void createIndexes(Executor tester) throws Throwable
             if (!skipColumns.contains(column.left))
             {
                 executeLocalIndexed(tester, String.format(template, column.left, indexedTable, column.left));
-                tester.waitForIndexQueryable(KEYSPACE, indexedTable);
+                tester.waitForTableIndexesQueryable(KEYSPACE, indexedTable);
```

Comment:
I think the original intention of this and the call on 304 was to wait for the single column index just created. In that case, we need to either use `waitForIndexQueryable(keyspace, indexName)` or move the `waitForTableIndexesQueryable()` outside the loop. The former might be better isolated. WDYT?

Reply:
Might require a change to `Executor`, but a small one...

Reply:
Yep, I've changed it to use `waitForIndexQueryable` and removed some minor code duplication.